### DISCUSSION TITLE
bug: PDF files above 1 MB cannot be uploaded into a chat and used

### DISCUSSION THREAD

**Discussion by YetheSamartaka** [open-webui/open-webui#3818] [General]
# Bug Report

## Description

**Bug Summary:**
PDF files above 1 MB cannot be drag and dropped/uploaded via + into a chat and used. I quickly found wiki article that upon dowloading has 3 MB, but the file I encountered that with had little above 1 MB. Files below 1 MB are working fine.

**Steps to Reproduce:**
Take a PDF file above 1 MB and try uploading it to Chat instance

Observe that it is not being uploaded

**Expected Behavior:**
Files above 1 MB should get uploaded and be usable the same way as PDFs below 1 MB

**Actual Behavior:**
Upon file upload above 1 MB upload, nothing happens.

## Environment

- **Open WebUI Version:** [e.g., 0.3.8]

- **Operating System:** [Ubuntu 20.04]
- **Browser (if applicable):** [Firefox, Chrome, Brave, Edge, Opera - seems to be happening on all]

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.

## Installation Method

Dockerfile

## Additional Information

I'm using text-embedding-3-large as an embedding model for documents

Not sure if it affects other filetypes.

These are the files that you can try. First will work, second won't.
[working_file.pdf](https://github.com/user-attachments/files/16190104/working_file.pdf)
[not_working.pdf](https://github.com/user-attachments/files/16190111/not_working.pdf)

**Comment by justinh-rahb** [comment]
Do you have a reverse proxy in front of your WebUI? Have you adjusted the max file upload size parameter in that? 1MB is a common default for some. WebUI's internal webserver has no such limit, so this seems to be something else at play here.

**Reply by YetheSamartaka** [reply]
As a matter of fact, I do. I will try raising the limit, thank you very much for the suggestion. If that is resolved, I will close this discussion.

**Reply by YetheSamartaka** [reply]
I've tried raising the limit and I still cannot upload the file. What other things I could try/investigate? @justinh-rahb

**Reply by justinh-rahb** [reply]
Max POST size? 🤔

**Reply by YetheSamartaka** [reply]
Oddly enough, increasing the limit fixed the issue but it took a while to properly take an effect. Issue solved, thank you very much.

**Reply by 0xGitGuy** [reply]
What code do you exactly use in the server config file? 
 
  I'm using `text-embedding-3-small`  As the embedding model. Nginx defualt config: `client_max_body_size 500M;`

 PDF files generally under 1MB get successfully uploaded when I drag and drop.  Anything above approximately 1 MB gets the spinning icon for a while and it just disappears. The physical file is uploaded to the server, but you don't see it in the open web UI chat. The file just disappears in the UI.

**Reply by Infinnerty** [reply]
I also have the same issue as @0xGitGuy (using a Gateway in GCP)

**Comment by scepterus** [comment]
I have both this bug, and this:
https://github.com/open-webui/open-webui/discussions/8007
Only files that are under 1MB are being referenced, and only the first file that is uploaded, so if I try to upload a file over 1MB, or if I try to upload a file that's a second file in a chat, they will both be ignored.

**Reply by rgaricano** [reply]
check upload size limits: adminSettings/Documents - Max Upload Size

Also, if you use some webserver (e.g. nginx) check the setup of upload limits.

**Reply by scepterus** [reply]
Size limit is empty, so should be unlimited. I use the openwebui docker with ollama, if that answers your second remark.

**Reply by rgaricano** [reply]
then maybe it's limitted in other place, what it's your document Content Extraction Engine?
(I can upload bigger size files without problems with defaults settings)

**Reply by scepterus** [reply]
My engine is the default. and the issue isn't that I can't upload, the files upload correctly, it's that the llm does not see them if they are larger than 1MB.
